### PR TITLE
feat: log error and stacktrace when panic in goroutine

### PR DESCRIPTION
**Pull request by qfrank**
relate [comment1](https://github.com/status-im/status-go/pull/5849#issuecomment-2360189471) / [comment2](https://github.com/status-im/status-go/pull/5849#issuecomment-2362701473)

relate [issue](https://github.com/status-im/status-mobile/issues/21126)

### CODE REVIEW

File: waku/v2/api/filter/filter.go [hunk 1]
```diff
@@ -213,12 +214,14 @@ func (apiSub *Sub) multiplex(subs []*subscription.SubscriptionDetails) {
 	for _, subDetails := range subs {
 		apiSub.subs[subDetails.ID] = subDetails
 		go func(subDetails *subscription.SubscriptionDetails) {
+			defer utils.LogOnPanicAndRethrow()
```
Comment (chaitanyaprem) [inline waku/v2/api/filter/filter.go:217]: There are many places in the code where go-routines are not started with `go func(` rather `go <func name>`. 
Maybe good to search with `go<space>` and replace them.
We will have to invoke this in such places by replacing it with a `go func(` style and with defer. 

some examples, but there are a lot more: 
- https://github.com/waku-org/go-waku/blob/798c9c5d819ac8f01cd77912dcd80a702c5ed3b5/waku/v2/api/filter/filter_manager.go#L85
- https://github.com/waku-org/go-waku/blob/798c9c5d819ac8f01cd77912dcd80a702c5ed3b5/waku/v2/api/filter/filter_manager.go#L103
- https://github.com/waku-org/go-waku/blob/798c9c5d819ac8f01cd77912dcd80a702c5ed3b5/waku/v2/api/filter/filter_manager.go#L138
- https://github.com/waku-org/go-waku/blob/798c9c5d819ac8f01cd77912dcd80a702c5ed3b5/waku/v2/api/filter/filter_manager.go#L231
- https://github.com/waku-org/go-waku/blob/798c9c5d819ac8f01cd77912dcd80a702c5ed3b5/waku/v2/node/wakunode2.go#L355

File: waku/v2/utils/logger.go [hunk 2]
```diff
@@ -81,3 +82,10 @@ func InitLogger(encoding string, output string, name string, level zapcore.Level
 
 	log = logging.Logger(name).Desugar()
 }
+
+func LogOnPanicAndRethrow() {
```
Comment (chaitanyaprem) [inline waku/v2/utils/logger.go:86]: function name can be aligned with changes in status-go

**Review by richard-ramos — APPROVED** [review]
LGTM

**Review by chaitanyaprem — APPROVED** [review]
LGTM

File: waku/v2/utils/logger.go [hunk 3]
```diff
@@ -81,3 +82,10 @@ func InitLogger(encoding string, output string, name string, level zapcore.Level
 
 	log = logging.Logger(name).Desugar()
 }
+
+func LogOnPanic() {
+	if err := recover(); err != nil {
+		Logger().Error("panic in goroutine", zap.Any("error", err), zap.String("stacktrace", string(debug.Stack())))
```
[the remark this answers is not in the document]
Comment (qfrank) [reply waku/v2/utils/logger.go:88]: emm.. err is `any` type here except convert it to error type first

**Pull request merged**